controllers/pulp/controller.go

```diff
@@ -223,15 +235,29 @@ func (r *PulpReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return pulpController, nil
 	}
 
-	if strings.ToLower(pulp.Spec.IngressType) == "route" {
-		log.V(1).Info("Running route tasks")
-		pulpController, err = r.pulpRouteController(ctx, pulp, log)
-		if err != nil {
-			return pulpController, err
-		} else if pulpController.Requeue {
-			return pulpController, nil
-		} else if pulpController.RequeueAfter > 0 {
-			return pulpController, nil
+	// if this is the first reconciliation loop (.status.ingress_type == "") OR
+	// if there is no update in ingressType field
+	if len(pulp.Status.IngressType) == 0 || pulp.Status.IngressType == pulp.Spec.IngressType {
+		if strings.ToLower(pulp.Spec.IngressType) == "route" {
+			log.V(1).Info("Running route tasks")
+			pulpController, err = r.pulpRouteController(ctx, pulp, log)
+			if err != nil {
+				return pulpController, err
+			} else if pulpController.Requeue {
+				return pulpController, nil
+			} else if pulpController.RequeueAfter > 0 {
+				return pulpController, nil
+			}
+		} else {
+			log.V(1).Info("Running web tasks")
+			pulpController, err = r.pulpWebController(ctx, pulp, log)
+			if err != nil {
+				return pulpController, err
+			} else if pulpController.Requeue {
+				return pulpController, nil
+			} else if pulpController.RequeueAfter > 0 {
+				return pulpController, nil
+			}
 		}
 	} else if strings.ToLower(pulp.Spec.IngressType) == "ingress" {
```

Comment:
I believe it should be under `if len(pulp.Status.IngressType) == 0 || pulp.Status.IngressType == pulp.Spec.IngressType`

Reply:
Oh ... I need to refactor the controller based on your ingress PR
Thank you!

Reply:
Actually, I'll open an issue for this because we still need to handle the ingress reconciliation.